mountpoint-s3-client/src/object_client.rs

```diff
@@ -1,14 +1,42 @@
 use async_trait::async_trait;
 use auto_impl::auto_impl;
 use futures::Stream;
-use std::{fmt, ops::Range};
+use std::str::FromStr;
+use std::{fmt, ops::Range, string::ParseError};
 use thiserror::Error;
 use time::OffsetDateTime;
 
 /// A single element of the [ObjectClient::get_object] response is a pair of offset within the
 /// object and the bytes starting at that offset.
 pub type GetBodyPart = (u64, Box<[u8]>);
 
+#[derive(Debug, Clone)]
+pub struct ETag {
+    etag: String,
+}
+
+impl ETag {
+    pub fn as_str(&self) -> &str {
+        &self.etag
+    }
+
+    pub fn for_tests() -> Self {
+        Self {
+            etag: "test_etag".to_string(),
+        }
+    }
```

Comment:
I did not add test because it was used in mock object which is later used in tests but itself is not within test. I can directly use `from_str()` in MockObject and compile it only for test?

Reply:
Yeah, `from_str()` should be for both main build and test; `for_tests()` tests only.

Is there an example section of that's an issue?

Reply:
```
impl<T: AsRef<[u8]>> From<T> for MockObject {
    fn from(bytes: T) -> Self {
        MockObject::from_bytes(bytes.as_ref(), ETag::for_tests())
    }
}
```
Lines 148 to 152 in mock_client.rs

Reply:
Also, in `prefetch_benchmark.rs` and reftests. But yes, I will call them using `from_str()`. And compile `for_tests()` only for tests.

Reply:
Ah, I see.

```
   Compiling mountpoint-s3 v0.2.0 (/Users/djonesoa/devel/mountpoint-s3/mountpoint-s3)
error[E0599]: no function or associated item named `for_tests` found for struct `object_client::ETag` in the current scope
   --> mountpoint-s3-client/src/mock_client.rs:150:54
    |
150 |         MockObject::from_bytes(bytes.as_ref(), ETag::for_tests())
    |                                                      ^^^^^^^^^ function or associated item not found in `object_client::ETag`
    |
   ::: mountpoint-s3-client/src/object_client.rs:14:1
    |
14  | pub struct ETag {
    | --------------- function or associated item `for_tests` not found for this struct

For more information about this error, try `rustc --explain E0599`.
error: could not compile `mountpoint-s3-client` due to previous error
```

Because `MockClient` is not test-only.

I'm not sure how best to update MockClient so instead, I'd suggest we just reduce the scope of `ETag::for_tests` from `pub` to `pub(crate)`

```suggestion
    pub(crate) fn for_tests() -> Self {
        Self {
            etag: "test_etag".to_string(),
        }
    }
```

Reply:
Ah, of course. It's not just used in `mountpoint-s3-client`, but also `mountpoint-s3`. Sorry! Happy to leave as is for now.